numpy/core/tests/test_getlimits.py

```diff
@@ -40,18 +41,22 @@ def test_singleton(self):
         ftype2 = finfo(longdouble)
         assert_equal(id(ftype), id(ftype2))
 
+def assert_finfo_equal(f1, f2):
+    # assert two finfo instances have the same attributes
+    for attr in ('bits', 'eps', 'epsneg', 'iexp', 'machep',
+                 'max', 'maxexp', 'min', 'minexp', 'negep', 'nexp',
+                 'nmant', 'precision', 'resolution', 'tiny',
+                 'smallest_normal', 'smallest_subnormal'):
+        assert_equal(getattr(f1, attr), getattr(f2, attr),
+                     f'finfo instances differ on {attr}')
```

Comment:
Might include the failing types in the message so we know what was tested and failed.
```
f"{f1} and {f2} differ on {attr}"
```